### PR TITLE
CAMS-450 - Sort attorneys

### PR DESCRIPTION
**Pull request by btposey**
# Purpose

Sort attorneys in the case assignment modal.

# Major Changes

* added a sort function to the attorney list

# Testing/Validation

* wrote a BDD style test
* manual test

# Definition of Done:

- [x] Acceptance criteria met - The system meets all acceptance criteria
- [x] Feature toggles created - Features that are deployed but not released have toggles
- [x] Design QA passed - All visual UI looks and works the way it’s supposed to
- [x] Usability validated - Team ensures the working code is easy to use
- [x] Accessibility tests passed - System passes automated and manual accessibility tests and supports applicable devices
- [x] Code refactored for clarity - Developers can understand the work simply by reviewing the code
- [x] Dependency Rule followed - More important code doesn’t directly depend on less important code
- [x] Development debt eliminated - UX and code aligns to the team’s latest understanding of the domain
- [ ] Source code merged - Code has been merged into the main branch
- [x] 90% unit test coverage achieved - Automated test coverage tools indicate coverage of >= 90%
- [ ] Code reviewed - Code is reviewed by at least two other team members before being merged
- [x] Code quality checks passed - Code passes all automated quality checks
- [x] Security scans passed - Code passes expected scans for vulnerabilities and compliance issues
- [x] Threat model updated - The threat model incorporates new threats and remediations
- [x] Build process updated - Automated build process include code supporting the user story
- [x] Load/performance tests passed - Performance tests include the functionality of the user story
- [x] Documents are updated - All required documentation is up to date and version controlled
- [ ] Code is deployed - Code is deployed to the highest production like environment

### CODE REVIEW

File: user-interface/src/test/staff-assignment.test.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      expect(staffAssignmentTab).toBeInTheDocument();
+      fireEvent.click(staffAssignmentTab!);
+    });
+
+    let firstCaseAssignmentButton;
+    await waitFor(async () => {
+      // TODO: Can we use a better selector than test ID since it is closer to an implementation detail than a descriptive label?
+      firstCaseAssignmentButton = await screen.queryByTestId('open-modal-button-0');
+      expect(firstCaseAssignmentButton).toBeInTheDocument();
+      fireEvent.click(firstCaseAssignmentButton!);
```
Comment (jamesobrooks) [inline user-interface/src/test/staff-assignment.test.tsx:37]: I think this should be either in a `.then()` or outside of the `waitFor` altogether.

**Review by jamesobrooks — COMMENTED** [review]
I'm generally okay with the approach, but I would note that it will require us to maintain the mock api where our more recent approach has been to use mocking.

**Comment by btposey** [comment]
The use of MockApi2 was intentional for the purpose of this test. I went with it because my mocking was basically doing the same as the existing mock API so I deferred writing another for now since this is a learning exercise.

**Comment by btposey** [comment]
It would be nice to consider a mock API next generation based on some of the in memory DB support we have in the codebase. The idea being we can load data for specific use cases into memory for the tests and so we can support write operations rather than read only. Once we have that mature capability we can retire MockApi2 to support unit tests and a11y tests.

**Comment by jamesobrooks** [comment]
> It would be nice to consider a mock API next generation based on some of the in memory DB support we have in the codebase. The idea being we can load data for specific use cases into memory for the tests and so we can support write operations rather than read only. Once we have that mature capability we can retire MockApi2 to support unit tests and a11y tests.

I'm curious how we would us mocking for a11y testing. The original use of the mock api was specifically to deal with running for a11y. I guess we would just make setting up the database part of the pa11y start script.

File: user-interface/src/test/staff-assignment.test.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    let staffAssignmentTab;
+    await waitFor(() => {
+      staffAssignmentTab = screen.queryByText('Staff Assignment');
+      expect(staffAssignmentTab).toBeInTheDocument();
+      fireEvent.click(staffAssignmentTab!);
+    });
+
+    let firstCaseAssignmentButton;
+    await waitFor(async () => {
+      // TODO: Can we use a better selector than test ID since it is closer to an implementation detail than a descriptive label?
```
Comment (fmaddenflx) [inline user-interface/src/test/staff-assignment.test.tsx:34]: We can use any selector we like with querySelector

Comment (btposey) [reply user-interface/src/test/staff-assignment.test.tsx:34]: I tried the text selector and that was no bueno.

**Review by jamesobrooks — APPROVED** [review]

**Review by amorrow-flexion — APPROVED** [review]

**Pull request merged**